src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -176,6 +179,23 @@ protected override void DetachCore()
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>Gets the highest header between the header of the last block that was sent to a peer and the expected peer's tip.</summary>
+        private ChainedHeader GetHighestHeader(INetworkPeer peer)
+        {
+            this.logger.LogTrace("({0}:'{1}')", nameof(peer), peer.RemoteSocketEndpoint);
+
+            var chainBehavior = peer.Behavior<ChainHeadersBehavior>();
```

Comment:
I tend to set `chainBehavior` as a field on the behaviour and set it on `AttachCore`

Reply:
Yeah. This whole behavior requires a refactoring, it's a big mess. 

Will fix this one.